src/2024h2/higher-level-rust.md

```diff
@@ -0,0 +1,283 @@
+
+### The status quo
+
+Rust has recently seen tremendous adoption in a number of high-profile projects. These include but are not limited to: Firecracker, Pingora, Zed, Datafusion, Candle, Gecko, Turbopack, React Compiler, Deno, Tauri, InfluxDB, SWC, Ruff, Polars, SurrealDB, NPM and more. These projects tend to power a particular field of development: SWC, React, and Turbopack powering web development, Candle powering AI/ML, Ruff powering Python, InfluxDB and Datafusion powering data science etc.
+
+These projects tend to focus on accelerating development in higher level languages. In theory, Rust itself would be an ideal choice for development in the respective spaces. A Rust webserver can be faster and more reliable than a JavaScript webserver. However, Rust's perceived difficulty, verbosity, compile times, and iteration velocity limit its adoption in these spaces. Various language constructs nudge developers to a particular program structure that might be non-obvious at its outset, resulting in slower development times. Other language constructs influence the final architecture of a Rust program, making it harder to migrate one's mental model as they transition to using Rust. Other Rust language limitations lead to unnecessarily verbose or noisy code. While Rust is not necessarily a bad choice for any of these fields, the analogous frameworks (Axum, Bevy, Dioxus, Polars, etc) are rather nascent and frequently butt up against language limitations.
+
+If we could make Rust a better choice for "higher level" programming - apps, games, UIs, webservers, datascience, high-performance-computing, scripting - then Rust would see much greater adoption outside its current bubble. This would result in more corporate interest, excited contributors, and generally positive growth for the language. With more "higher level" developers using Rust, we might see an uptick in adoption by startups, research-and-development teams, and the physical sciences which could lead to more general innovation.
+
+While new languages focused on high-level applications are gaining traction, the thesis of this project goal is that Rust itself can be tweaked to make it a better choice overall.
+
+Generally we believe this boils down two focuses:
+
+- Make Rust programs faster to write
+- Shorten the iteration cycle of a Rust program
+
+### The next few steps
+
+The two key areas we've identified as places to cut down on verbosity and make Rust easier to work are:
+
+- Reducing the frequency of an explicit ".clone()" for cheaply cloneable items
+- Partial borrows for structs
+
+The key areas we've identified as avenues to speed up iterative development include:
+
+- Speeding up or caching proc macro expansion
+- A per-user cache for compiled artifacts
```

Comment:
What does this mean? Something like a global `target` directory, so that you don't have to recompile `abc v0.1.0` if you already compiled it elsewhere?